### DISCUSSION TITLE
What is the best way to listen to database changes on an encrypted table?

### DISCUSSION THREAD

**Discussion by nirmal-kc** [supabase/supabase#20129] [Questions] [answered]
I have a simple chat messages table "chat_messages" where the column "message" is encrypted. The decrypted value is only available on "decrypted_chat_messages" view. I want to add a stream/subscribe to changes in the table. Currently, I am listening to the changes on table "chat_messages", fetching the ID of the newly inserted record, and again making an API call to fetch the decrypted value from the view. This works, however, for every new message, 1 additional DB call is made to fetch the value of "decrypted_message". 
Is there a better approach?

**Comment by GaryAustin1** [comment, marked as answer]
Realtime does not work on views so that sounds like the best approach unless you come up with your own way to do an encrypted table with the column encrypted/decrypted thru some method on the client.
